src/lib.rs

```diff
@@ -540,12 +545,20 @@ impl LightningNode {
     /// Decode a user-provided string (usually obtained from QR-code or pasted).
     pub fn decode_data(&self, data: String) -> std::result::Result<DecodedData, DecodeDataError> {
         match self.rt.handle().block_on(parse(&data)) {
-            Ok(InputType::Bolt11 { invoice }) => Ok(DecodedData::Bolt11Invoice {
-                invoice_details: InvoiceDetails::from_ln_invoice(
-                    invoice,
-                    &self.get_exchange_rate(),
-                ),
-            }),
+            Ok(InputType::Bolt11 { invoice }) => {
+                if map_breez_network_into_network(invoice.network) != self.environment.network {
+                    return Err(DecodeDataError::InvalidInput {
```

Comment:
I think unsupported is a good candidate here (can even include the type of network the invoice had).